sql/slave.cc

```diff
@@ -4161,9 +4168,22 @@ static int exec_relay_log_event(THD* thd, Relay_log_info* rli,
 
       In parallel replication, we might queue a large number of events, and
       the user might be surprised to see a claim that the slave is up to date
-      long before those queued events are actually executed.
+      long before those queued events are actually executed. The check for
+      parallel.workers_idle() ensures all events on the parallel replica have
+      been executed (note that it always returns TRUE in the serial case).
+
+      rli->sql_thread_caught_up is negated here to ensure that the value of
+      Seconds_Behind_Master in SHOW SLAVE STATUS is consistent with the update
+      of last_master_timestamp. It was previously unset immediately after
+      reading an event from the relay log; however, for the duration between
+      that unset and the time that LMT would be updated could lead to spikes
+      in SBM.
      */
-    if ((!rli->mi->using_parallel()) && event_can_update_last_master_timestamp(ev))
+    if ((unlikely(rli->last_master_timestamp == 0) ||
```

Comment:
As it pointed earlier `rli->last_master_timestamp == 0` must fall under `rli->last_master_timestamp < ev->when`, *and* (now it got clear to me) at that point `queued_count == dequeued_count` we could remove l.4182 disjunction argument or convert into `assert(lmt != 0 | queued == dequeued)`. For safely I suggest to do so in 11.3.